tests/unit_tests/test_ladder_service.py

```diff
@@ -93,25 +95,35 @@ async def test_load_from_database_new_data(ladder_service, database):
     test_queue.find_matches.assert_called()
 
 
+@pytest.mark.filterwarnings("ignore:.*uses the '.*' fixture")
+@given(
+    player1=st_players(player_id=1),
+    player2=st_players(player_id=2)
+)
+@settings(deadline=300)
 async def test_start_game_1v1(
     ladder_service: LadderService,
     game_service: GameService,
-    player_factory,
     monkeypatch,
+    player1,
+    player2
 ):
     queue = ladder_service.queues["ladder1v1"]
-    p1 = player_factory("Dostya", player_id=1, with_lobby_connection=True)
-    p2 = player_factory("Rhiza", player_id=2, with_lobby_connection=True)
 
     monkeypatch.setattr(LadderGame, "wait_hosted", CoroutineMock())
     monkeypatch.setattr(LadderGame, "wait_launched", CoroutineMock())
-    await ladder_service.start_game([p1], [p2], queue)
+    monkeypatch.setattr(LadderGame, "timeout_game", CoroutineMock())
+    for player in (player1, player2):
+        player.__owned_lobby_connection = mock.Mock(launch_game=CoroutineMock())
```

Comment:
This is different from the autospec mock in`test/conftest.py`. Would that one not work? Maybe it would make sense to allow to set `with_lobby_connection` also for players drawn from `st_players`?

Reply:
It would work and you could use it already since kwargs are forwarded. The problem is that autospecing is expensive and tends to cause the test to exceed the default execution deadline of 200ms. 

Maybe I should use a spec_set here instead of a full autospec. I think that would be fast and still give most of the benefits of the autospec'd mock.